movie/src/main/java/com/calvinnor/movie/commons/util/NetworkUtils.kt

```diff
@@ -0,0 +1,52 @@
+package com.calvinnor.movie.commons.util
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import android.os.Build
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+
+object NetworkUtils : ConnectivityManager.NetworkCallback() {
+
+    private val networkLiveData: MutableLiveData<Boolean> = MutableLiveData()
+
+    fun getNetworkLiveData(context: Context): LiveData<Boolean> {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            connectivityManager.registerDefaultNetworkCallback(this)
+        } else {
+            val builder = NetworkRequest.Builder()
```

Comment:
Nit: You don't need to have a `builder` variable. Just create it and build on the spot 

```
connectivityManager.registerNetworkCallback(NetworkRequest.Builder().build(), this)
```